### PR TITLE
914: Split up translation-row template for easier modification by plugins

### PR DESCRIPTION
**Pull request by superdav42**
This will make maintenance easier for some plugins which need to override the entire translation-row template. Now the can just override the smaller templates or take advantage of the new filters and actions. 
For #914

### CODE REVIEW

**Comment by toolstack** [comment]
Is there a specific use-case for this?

Also, please open an issue describing the change that we can track against.

**Comment by superdav42** [comment]
Made a few changes to bring things up to coding standard and added a few more logical templates.

For use cases the [plugin](https://github.com/superdav42/gp-reject-feedback ) I was working on to add reject reasons to meta status section could really benefit from this. And translate.wordpress.com uses the actions introduced in this PR. If these project/plugins could use it future plugins likely could as well. Also the core code will be easier to maintain and understand with things split up more like this.

File: gp-templates/translations.php [hunk 1]
```diff
@@ -26,6 +26,17 @@
...
+
+/**
+ * Fires header before toolbar
+ *
+ * @since 3.0.0
+ *
+ * @param GP_Translation_Set $translation_set The current translation set.
+ * @param GP_Project $project The current project.
+ */
+do_action( 'gp_translation_view_message', $translation_set, $project );
```
Comment (ocean90) [inline gp-templates/translations.php]: This seems unrelated and should be part of a separate PR.

File: gp-templates/translation-row-editor.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Template for the editor part of a single translation row in a translation set display
+ *
+ * @package    GlotPress
+ * @subpackage Templates
+ */
+
+$colspan = apply_filters( 'gp_translation_row_editor_colspan', $can_approve ? 5 : 4 );
```
Comment (ocean90) [inline gp-templates/translation-row-editor.php]: Filter docs are missing.

File: gp-templates/translation-row-editor.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+					<?php endforeach; ?>
+				<?php endif; ?>
+			<?php endif; ?>
+			<?php gp_tmpl_load( 'translation-row-editor-actions', get_defined_vars() ); ?>
+		</div>
+		<?php gp_tmpl_load( 'translation-row-editor-meta', get_defined_vars() ); ?>
+	</td>
+	<?php
+	/**
+	 * Fires after editor column
```
Comment (ocean90) [inline gp-templates/translation-row-editor.php]: Descriptions should end with a full stop.

File: gp-templates/translation-row-editor.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	 *
+	 * @since 3.0.0
+	 *
+	 * @param GP_Translation $t The current translation.
+	 * @param GP_Translation_Set $translation_set The current translation set.
```
Comment (ocean90) [inline gp-templates/translation-row-editor.php]: The variables should be aligned.

Comment (ocean90) [reply gp-templates/translation-row-editor.php]: (Wondering if should take the opportunity to rename `$t` to `$translation` for clarity.)

**Review by ocean90 — CHANGES_REQUESTED** [review]
Beside the new filter this looks good to me.

File: gp-templates/translations.php (hunk 1 above)
Comment (superdav42) [reply gp-templates/translations.php]: You're right.

File: gp-templates/translation-row-editor.php (hunk 4 above)
Comment (superdav42) [reply gp-templates/translation-row-editor.php]: Why not. I'm a fan of descriptive variable names.

**Comment by superdav42** [comment]
@ocean90 All issues addressed.

**Review by ocean90 — APPROVED** [review]
Looks good to me.

**Review by toolstack — COMMENTED** [review]
My only concern with this is that, whenever we make significant changes to the templates we tend to break sites that use a customs template directory.  I'm ok with it for this as we can do it in 3.0 which is a major release so users with customizations should be paying more attention anyway so it should be manageable.

File: gp-templates/translation-row.php [hunk 5]
```diff
@@ -6,248 +6,36 @@
...
-			</dl>
-		</div>
-	</td>
-</tr>
+gp_tmpl_load(
+	'translation-row-preview', array(
+		'translation'             => $translation,
+		'can_edit'                => $can_edit,
+		'can_approve'             => $can_approve,
+		'can_approve_translation' => $can_approve_translation,
```
Comment (toolstack) [inline gp-templates/translation-row.php]: I'd suggest passing in all of the variables that were available in the old unified template here, in case a plugin was expecting them.

File: gp-templates/translation-row.php [hunk 6]
```diff
@@ -6,248 +6,36 @@
...
+		'translation'             => $translation,
+		'can_write'               => $can_write,
+		'can_edit'                => $can_edit,
+		'can_approve'             => $can_approve,
+		'can_approve_translation' => $can_approve_translation,
+		'can_edit'                => $can_edit,
+		'locale'                  => $locale,
+		'project'                 => $project,
+		'translation_set'         => $translation_set,
+	)
```
Comment (toolstack) [inline gp-templates/translation-row.php]: I'd suggest passing in all of the variables that were available in the old unified template here, in case a plugin was expecting them.

File: gp-templates/translation-row-editor.php [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+ *
+ * @since 3.0.0
+ *
+ * @param int $colspan The colspan of editor column.
+ */
+$colspan = apply_filters( 'gp_translation_row_editor_colspan', $can_approve ? 5 : 4 );
+
+$singular = sprintf(
+	/* translators: %s: Original singular form of the text */
+	__( 'Singular: %s', 'glotpress' ),
```
Comment (toolstack) [inline gp-templates/translation-row-editor.php]: You can't include the 'Singluar:' part of the string here otherwise when we call `prepare_original()` later on it will get included in the markup, which we don't want.  For example of "Singular" was defined in a glossary then the "Singluar" before the colon would get get highlighted.

File: gp-templates/translation-row-editor.php [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+	'<span class="original">' . ( isset( $translation->singular_glossary_markup ) ? $translation->singular_glossary_markup : esc_translation( $translation->singular ) ) . '</span>'
+);
+$plural = sprintf(
+	/* translators: %s: Original plural form of the text */
+	__( 'Plural: %s', 'glotpress' ),
```
Comment (toolstack) [inline gp-templates/translation-row-editor.php]: Same as above.

File: gp-templates/translation-row.php (hunk 6 above)
Comment (superdav42) [reply gp-templates/translation-row.php]: The old template was using `get_defined_vars()` which seems like a bad idea so I changed it. Since we are targeting a major release I think it would be fine to explicitly pass the vars. We could also document the passed vars at the top of the template using @var. I see now that  `get_defined_vars()` is used in the other templates so I should change there too.

Comment (toolstack) [reply gp-templates/translation-row.php]: `get_defined_vars()` is used throughout GP in this manner, and while not ideal, I think maintaining consistency with the rest of the code base is probably better than changing it in just one location.

Perhaps for 4.0 we could look at changing this for all of GP.

**Comment by superdav42** [comment]
@toolstack Updated code to address your concerns. `get_defined_vars()` is now used to pass vars to templates and translation-row-editor.php shouldn't pass anything extra to prepare_original
